react-js-gmp/src/components/Counter.test.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import Counter from './Counter';
+
+describe('Counter Component', () => {
+  test('renders initial value provided in props', () => {
+    render(<Counter initialValue={5} />);
+    const counterValue = screen.getByText('Count: 5');
+    expect(counterValue).toBeInTheDocument();
+  });
+
+  test('clicking "decrement" button decrements the displayed value', () => {
+    render(<Counter initialValue={5} />);
+    const decrementButton = screen.getByText('Decrement');
+    fireEvent.click(decrementButton);
```

Comment:
in most cases it is better to use  `user-events` instead of `fireEvent` unless you specifically need a single DOM event to be fired. Pls check https://testing-library.com/docs/user-event/intro